qiskit/pulse/parameter_manager.py

```diff
@@ -424,7 +425,24 @@ def _unroll_param_dict(
                     )
                 out.update(zip(parameter, value))
             elif isinstance(parameter, str):
-                out[self.get_parameters(parameter)] = value
+                for param in self.parameters:
```

Comment:
Assignment by string was in fact not supported before. It did not raise an error, but it didn't do proper parameter assignment.

Reply:
I realised that if you try the following code with the current version in the main (that I had implemented) you have an error, yielding the need to really support the string assignment:


`a = Parameter("a")
with pulse.build(name="test") as sched:
    pulse.play(pulse.Gaussian(100, a, 50), pulse.DriveChannel(0))`

Moreover, if you remove the part in the `_unroll_dict`method the special case (elif) for string handling (leaving the default case), you end up with the parameter not being assigned at all.

Reply:
Hmm, that code doesn't error for me on `main`. Am I missing something from your example?

```python
a = Parameter("a")
with pulse.build(name="test") as sched:
    pulse.play(pulse.Gaussian(100, a, 50), pulse.DriveChannel(0))
```

However,

```python
sched.assign_parameters({"a": 1}).blocks
```

prints

    (Play(Gaussian(duration=100, sigma=50, amp=a, angle=0.0), DriveChannel(0)),)

So you are right about string assignment not working now. My previous test did not work properly because I tested `{a: 1}` and then `{"a": 1}` and did not account for the fact that the default is `inplace=True`.

Reply:
The code should have raised an error for the following reason: 
On the main, when you currently assign by the parameter by its name, the line 427 of the parameter_manager.py is failing: `out[self.get_parameters(parameter)] = value`.
In fact, the error is due to the fact that get_parameters() returns a list of parameters and not a parameter instance directly, making it an unusable key for the dict out. This is why we need to correct this with this PR.
Make sure you don't have the similar case as you depicted where the parameter has already been assigned (returning an empty list) for your test.

Reply:
Ah, okay. I was just a little confused by your example above. I think you mean to include `assign_parameters({"a": 1})` in it? I do get the error with string assignment on `main` now. I think my `main` was old before.